src/Azure/Orleans.Transactions.AzureStorage/Storage/AzureTransactionLogStorage.cs

```diff
@@ -359,34 +360,40 @@ private List<CommitRecord> DeserializeCommitRecords(byte[] serializerCommitRecor
 
         public class ArchivalRow : TableEntity
         {
+            public static string MinRowKey = MakeRowKey(long.MinValue);
+            public static string MaxRowKey = MakeRowKey(long.MaxValue);
+
+            public static string MakeRowKey(long firstTransactionId)
+            {
+                return $"arch_{CommitRow.ToRowKey(firstTransactionId)}";
```

Comment:
With this pattern one could change the commit row, without knowing they are also changing the archive row unintentionally.  I suggest this entity be authoritative of it's rowkey, and not delegate any part of it to the commit row.